### DISCUSSION TITLE
Page reloads completely upon refresh_token updates

### DISCUSSION THREAD

**Discussion by ilyabo** [supabase/supabase#5437] [Questions]
I noticed that my app's page reloads completely every now and then. It seems that it happens when Supabase is trying to refresh the auth token by fetching from this URL:
https://****.supabase.co/auth/v1/token?grant_type=refresh_token
But why does it lead to complete page reloads? Is it by design or am I doing something wrong?

Would appreciate any help on this issue!

**Comment by silentworks** [comment]
The supabase libraries don't reload your pages as they are so this is likely something else in your code doing this. It would help more if you could share your code or an example of where the error exists.

**Reply by ilyabo** [reply]
Thanks for the response! That's good to know. 
My app is based on this starter kit: https://github.com/vercel/nextjs-subscription-payments/
However, I cannot yet tell if it happens with the starter kit itself or only in my app.
But I am using basically the same user auth logic:
https://github.com/vercel/nextjs-subscription-payments/blob/main/utils/useUser.tsx#L38
